src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,705 @@ static ColliderJntSphInit D_80A17040 = {
+
+    // If touching wall or not in water, turn back and slow down for one frame.
+    if ((this->actor.bgCheckFlags & 8) || !(this->actor.bgCheckFlags & 0x20)) {
+        this->actor.home.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos);
+        this->actor.speedXZ *= 0.5f;
+    }
+
+    Math_StepToAngleS(&this->actor.world.rot.x, 0, 1500);
+    Math_StepToAngleS(&this->actor.world.rot.y, this->actor.home.rot.y, 3000);
+    Math_StepToAngleS(&this->actor.world.rot.z, 0, 1000);
+
+    this->actor.shape.rot = this->actor.world.rot;
+
+    // Raise if on a floor.
+    if (this->actor.bgCheckFlags & 1) {
+        Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y - 4.0f, 2.0f);
+    } else {
+        Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y - 10.0f, 2.0f);
+    }
+
+    // Shrink when close to disappearing.
+    if (this->timer < 100) {
+        Actor_SetScale(&this->actor, this->actor.scale.x * 0.982f);
+    }
+
+    this->skelAnime.playSpeed = CLAMP_MAX((this->actor.speedXZ * 1.5f) + 1.0f, 4.0f);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer <= 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+// Unique type functions
+
+void EnFish_Unique_SetupSwimIdle(EnFish* this) {
+    this->actor.gravity = 0.0f;
+    this->actor.minVelocityY = 0.0f;
+    this->timer = Rand_S16Offset(5, 35);
+    this->unk_250 = 0;
+    EnFish_SetInWaterAnimation(this);
+    this->actionFunc = EnFish_Unique_SwimIdle;
+}
+
+void EnFish_Unique_SwimIdle(EnFish* this, GlobalContext* globalCtx) {
+    static f32 speedStopping[] = { 0.0f, 0.04f, 0.09f };
+    static f32 speedMoving[] = { 0.5f, 0.1f, 0.15f };
+    f32 playSpeed;
+    u32 frames = globalCtx->gameplayFrames;
+    f32* speed;
+    s32 pad2;
+    f32 extraPlaySpeed;
+    s32 pad3;
+
+    if (this->actor.xzDistToPlayer < 60.0f) {
+        if (this->timer < 12) {
+            speed = speedMoving;
+        } else {
+            speed = speedStopping;
+        }
+    } else {
+        if (this->timer < 4) {
+            speed = speedMoving;
+        } else {
+            speed = speedStopping;
+        }
+    }
+
+    EnFish_SetYOffset(this);
+    Math_SmoothStepToF(&this->actor.speedXZ, speed[0], speed[1], speed[2], 0.0f);
+
+    extraPlaySpeed = 0.0f;
+
+    if ((EnFish_XZDistanceSquared(&this->actor.world.pos, &this->actor.home.pos) > SQ(15.0f))) {
+        if (!Math_ScaledStepToS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos),
+                                200)) {
+            extraPlaySpeed = 0.5f;
+        }
+    } else if ((this->timer < 4) && !Math_ScaledStepToS(&this->actor.world.rot.y, frames * 0x80, 100)) {
+        extraPlaySpeed = 0.5f;
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    playSpeed = (this->actor.speedXZ * 1.2f) + 0.2f + extraPlaySpeed;
+    this->skelAnime.playSpeed = CLAMP(playSpeed, 1.5f, 0.5);
```

Comment:
is no f required here?
```suggestion
    this->skelAnime.playSpeed = CLAMP(playSpeed, 1.5f, 0.5f);
```

Reply:
It is.